woodwork/tests/test_utils.py

```diff
@@ -566,3 +569,452 @@ def test_parse_logical_type_errors():
+
+
+def test_concat_cols_different_schema(sample_df):
+    df1 = sample_df[['id', 'age']]
+    df1.ww.init(index='id', logical_types={'id': 'AgeNullable'})
+    df2 = sample_df[['id', 'signup_date']]
+    df2.ww.init(index='id', logical_types={'id': 'IntegerNullable'})
+
+    assert df2.ww.schema != df1.ww.schema
+
+    combined_df_1 = concat([df1.ww.copy(), df2.ww.copy()], axis=1)
+    combined_df_1.ww.logical_types['id'] == AgeNullable
+
+    combined_df_2 = concat([df2.ww.copy(), df1.ww.copy()], axis=1)
+    combined_df_2.ww.logical_types['id'] == IntegerNullable
+
+
+def test_concat_rows_different_columns(sample_df):
+    sample_df.ww.init(time_index='signup_date')
+    df1 = sample_df.ww[['id', 'signup_date', 'full_name', 'email']]
+    df2 = sample_df.ww[['id', 'signup_date', 'phone_number', 'age', 'is_registered']]
+    # should work if dtypes dont change
+
+    combined_df = concat([df1, df2], axis=0)
+    assert combined_df.ww.schema == sample_df.ww.schema
+    assert to_pandas(combined_df).equals(pd.concat([to_pandas(df1), to_pandas(df2)]))
+
+
+def test_concat_combine_metadatas(sample_df):
+    df1 = sample_df[['id', 'full_name', 'email']]
+    df1.ww.init(table_metadata={'created_by': 'user0', 'test_key': 'test_val1'}, column_metadata={'id': {'interesting_values': [1, 2]}})
+    df2 = sample_df[['phone_number', 'age', 'signup_date', 'is_registered']]
+    df2.ww.init(table_metadata={'table_type': 'single', 'test_key': 'test_val2'}, column_metadata={'age': {'interesting_values': [33]}})
+
+    combined_df_1 = concat([df1, df2])
+
+    # The first table sets the metadata when there's overlap
+    assert combined_df_1.ww.metadata == {'created_by': 'user0',
+                                         'table_type': 'single',
+                                         'test_key': 'test_val1'}
+    assert combined_df_1.ww.columns['id'].metadata == {'interesting_values': [1, 2]}
+    assert combined_df_1.ww.columns['age'].metadata == {'interesting_values': [33]}
+
+    combined_df_2 = concat([df2, df1])
+
+    assert combined_df_2.ww.metadata == {'created_by': 'user0',
+                                         'table_type': 'single',
+                                         'test_key': 'test_val2'}
+    assert combined_df_2.ww.columns['id'].metadata == {'interesting_values': [1, 2]}
+    assert combined_df_2.ww.columns['age'].metadata == {'interesting_values': [33]}
+
+
+@patch("woodwork.table_accessor._validate_accessor_params")
+def test_concat_rows_validate_schema(mock_validate_accessor_params, sample_df):
+    sample_df.ww.init(validate=False)
+    copy_df = sample_df.ww.copy()
+
+    assert not mock_validate_accessor_params.called
+
+    concat([sample_df, copy_df], axis=0, validate_schema=False)
+
+    assert not mock_validate_accessor_params.called
+
+    concat([sample_df, copy_df], axis=0, validate_schema=True)
+
+    assert mock_validate_accessor_params.called
+
+
+def test_concat_cols_mismatched_index_adds_nans(sample_df):
```

Comment:
Added a test that does add multiple nans to the index column, causing an error to be raised for pandas dataframes because the index is no longer unique. It's still just being caught by woodwork init, though.